consensus/propagation/propagator.go

```diff
@@ -0,0 +1,16 @@
+package propagation
+
+import (
+	"github.com/tendermint/tendermint/libs/bits"
+	"github.com/tendermint/tendermint/p2p"
+	"github.com/tendermint/tendermint/types"
+)
+
+// Propagator provides the necessary propagation mechanism for the
+// consensus reactor and state.
+type Propagator interface {
+	GetProposal(height int64, round int32) (*types.Proposal, *types.PartSet, *bits.BitArray, bool)
+	ProposeBlock(proposal *types.Proposal, haves *bits.BitArray)
```

Comment:
could be a flup but `ProposeBlock` atm should be include the `proposal`, both partSets, and probably the hashes